jupyterhub_moss/spawner.py

```diff
@@ -290,35 +292,24 @@ async def create_options_form(spawner):
     _MEM_REGEXP = re.compile("^[0-9]*([0-9]+[KMGT])?$")
 
     def __validate_options(self, options):
-        """Check validity of options"""
+        """Check validity/syntax of options"""
         assert "partition" in options, "Partition information is missing"
         assert options["partition"] in self.partitions, "Partition is not supported"
 
-        partition_info = self.partitions[options["partition"]]
-
         if "runtime" in options:
-            runtime = parse_timelimit(options["runtime"])
-            assert (
-                runtime.total_seconds() <= partition_info["max_runtime"]
```

Comment:
Checks using partition info retrieved from `sinfo` like `max_runtime` are moved to `start`.